src/Niv/Git/Cmd.hs

```diff
@@ -98,6 +100,13 @@ parseGitPackageSpec =
               <> Opts.short 'b'
               <> Opts.metavar "BRANCH"
           )
+    parseRollback =
+      ("rollback-protection",) . Aeson.Bool
+        <$> Opts.flag'
+          True
```

Comment:
I'm thinking most people don't care too much about rollback protection (and may every now and then need to rollback). Does it make sense to enable this by default and impose a full extra `git clone` for every `git` source?

Reply:
I see why this is confusing but this is not enabled by default.
The code only enables it if `--rollback-protection` is given.

One could similarly write a flag that disables it when `--no-rollback-protection` is given, but I didn't see the need.